### PR TITLE
Remove PSP from api-gatway

### PR DESCRIPTION
**Pull request by triffer**
**Description**

Changes proposed in this pull request:

- Remove PSP
- Add restricted level security context configuration

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/15916

### CODE REVIEW

File: resources/api-gateway/templates/webhook-certificates.yaml [hunk 1]
```diff
@@ -8,12 +8,17 @@ metadata:
...
-  replicas: 1
   template:
     metadata:
       annotations:
         sidecar.istio.io/inject: "false"
     spec:
+      securityContext:
+        runAsUser: 65534
+        runAsGroup: 65534
+        runAsNonRoot: true
```
Comment (cnvergence) [inline resources/api-gateway/templates/webhook-certificates.yaml:19]: Shall we also include `readOnlyRootFilesystem: true`?

File: resources/api-gateway/values.yaml [hunk 2]
```diff
@@ -16,10 +16,11 @@ deployment:
     capabilities:
       drop:
       - ALL
-    privileged: false
     runAsGroup: 65534
     runAsNonRoot: true
     runAsUser: 65534
+    seccompProfile:
+      type: RuntimeDefault
```
Comment (cnvergence) [inline resources/api-gateway/values.yaml:23]: Shall we also include `readOnlyRootFilesystem: true?`

File: resources/api-gateway/templates/webhook-certificates.yaml (hunk 1 above)
Comment (triffer) [reply resources/api-gateway/templates/webhook-certificates.yaml:19]: Thanks, good finding! It's not relevant for Pod Security Admission, but still we should have it.

**Review by cnvergence — APPROVED** [review]
lgtm

**Pull request merged**